Automattic/wp-calypso
# Reader: show single site blocked message for site and feed streams

Fixes https://github.com/Automattic/wp-calypso/issues/3669.

Currently, when a site is blocked, all post cards for that site turn into a 'blocked' card. This works reasonably well on mixed streams with content from lots of sites, but it's a pretty poor experience on site and feed streams, which end up looking like this:

![34960462-c2660cfc-f9ef-11e7-8071-9bb64cb634f6](https://user-images.githubusercontent.com/17325/37110936-39cd3734-2236-11e8-95e5-8cf25aa2fa2f.png)

This PR changes it to look like this:

![screen shot 2018-03-08 at 14 28 38](https://user-images.githubusercontent.com/17325/37156367-d3e2b0b8-22dd-11e8-84fb-b659433be44b.png)

### Notes

I originally tried a hybrid of feed header and blocked message, but it didn't look great.

The illustration seemed the most neutral of those available - what do you think @jancavan?

When loading the site or feed stream afresh, you may see a flash of the site's posts while we're waiting for the API response. We can partially address this in the future by not returning posts for sites the user has blocked, but ideally we'd sync site blocks in the same way we currently do for follows. This would mean a new endpoint to return current site blocks.

### To test

Try blocking a feed like http://calypso.localhost:3000/read/feeds/59175237 and make sure the blocked message appears as above.

The corresponding site stream for the feed should also show the blocked message: http://calypso.localhost:3000/read/blogs/122301668.
